sound/soc/soc-dai.c

```diff
@@ -617,8 +617,14 @@ int snd_soc_pcm_dai_prepare(struct snd_pcm_substream *substream)
 static int soc_dai_trigger(struct snd_soc_dai *dai,
 			   struct snd_pcm_substream *substream, int cmd)
 {
+	struct snd_soc_pcm_stream *stream;
```

Comment:
commit message: 

ASoC: soc-dai: don't call trigger ops if the stream is not valid

soc_dai_trigger may be called when the stream is not valid
We should not call the trigger ops in that case to avoid unexpected
behavior or unintended side effects.